tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -259,9 +259,14 @@ public bool AsBool(bool defaultValue, Func<bool, bool>? validator)
                 return value;
             }
 
-            // Horrible that we have to stringify the dictionary, but that's all that's available in the telemetry api
-            // _keys.Telemetry.Record(_keys.Key, string.Join(", ", defaultValue.Select(kvp => $"{kvp.Key}:{kvp.Value}")), ConfigurationOrigins.Default);
-            // return defaultValue;
+            if (getDefaultValue != null)
+            {
+                // Horrible that we have to stringify the dictionary, but that's all that's available in the telemetry api
+                var defaultValue = getDefaultValue();
+                Telemetry.Record(Key, string.Join(", ", defaultValue.Select(kvp => $"{kvp.Key}:{kvp.Value}")), true, ConfigurationOrigins.Default);
```

Comment:
Random thought - we could add an additional argument to supply the pre-stringified default value to avoid doing the horrible concatenation?

Alternatively, given we just return an empty dictionary in all these cases, we could check for that first and just use `string.Empty`?

Reply:
`string.Join` already has a path to return `string.Empty` if there are no elements

Reply:
Meh I suppose we'd save the enumerator

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/5126/commits/7d4b6a84d089ba85674e50047ce60bc18a403cfc